src/hints/hintSchema.ts

```diff
@@ -0,0 +1,201 @@
+  .transform((object) => ({
+    type: OpType.Deref,
+    cell: object.Deref,
+  }));
+
+/** Zod object to parse DoubleDeref */
+const doubleDeref = z
+  .object({
+    DoubleDeref: z.object({
+      Deref: cellRef,
+      offset: z.number(),
+    }),
+  })
+  .transform((object) => ({
+    type: OpType.DoubleDeref,
+    cell: object.DoubleDeref.Deref,
+    offset: object.DoubleDeref.offset,
+  }));
+
+/** Zod object to parse Immediate */
+const immediate = z
+  .object({
+    Immediate: z.string().transform((value) => new Felt(BigInt(value))),
+  })
+  .transform((object) => ({ type: OpType.Immediate, value: object.Immediate }));
+
+/**
+ * Operation that can be done by a BinOp
+ */
+export enum Operation {
+  Add = 'Add',
+  Mul = 'Mul',
+}
+
+/** Zod object to parse a BinOp */
+const binOp = z
+  .object({
+    BinOp: z.object({
+      op: z.enum(['Add', 'Mul']).transform((op) => {
+        return op === 'Add' ? Operation.Add : Operation.Mul;
+      }),
+      a: cellRef,
+      b: z.union([deref, immediate]),
+    }),
+  })
+  .transform((object) => ({
+    type: OpType.BinOp,
+    op: object.BinOp.op,
+    a: object.BinOp.a,
+    b: object.BinOp.b,
+  }));
+
+/** Zod object to parse a ResOp */
+export const resOp = z.union([deref, doubleDeref, immediate, binOp]);
+
+/**
+ * A CellRef is an object defining a register and an offset
+ *
+ * `register + offset`
+ */
+export type CellRef = z.infer<typeof cellRef>;
+
+/**
+ * A Deref is a CellRef wrapped in a `Deref` field
+ *
+ * `[register + offset]`
+ */
+export type Deref = z.infer<typeof deref>;
+
+/**
+ * A DoubleDeref is a Deref wrapped in a `DoubleDeref` field
+ * with a second offset
+ *
+ * `[[register + offset] + offset]`
+ */
+export type DoubleDeref = z.infer<typeof doubleDeref>;
+
+/**
+ * An Immediate is an hex string representing a Felt,
+ * wrapped in an `Immediate` field
+ *
+ * `0x1000`
+ */
+export type Immediate = z.infer<typeof immediate>;
+
+/**
+ * A BinOp is an addition or a multiplication
+ * between two values.
+ * - `a` is a cellRef which should points to a Felt
+ * - `b` is either a Deref or an Immediate, should be a Felt in both cases.
+ */
+export type BinOp = z.infer<typeof binOp>;
+
+/** A ResOp is either a Deref, DoubleDeref, Immediate or BinOp */
+export type ResOp = z.infer<typeof resOp>;
+
+/** Name to identify which hint is executed  */
+export enum HintName {
+  AllocSegment = 'AllocSegment',
+  TestLessThan = 'TestLessThan',
+}
+
+/** Zod object to parse AllocSegment hint */
+export const allocSegment = z
+  .object({ AllocSegment: z.object({ dst: cellRef }) })
+  .transform(({ AllocSegment: { dst } }) => ({
+    type: HintName.AllocSegment,
+    dst,
+  }));
+
+/** Zod object to parse TestLessThan hint */
+export const testLessThan = z
+  .object({
+    TestLessThan: z.object({ lhs: resOp, rhs: resOp, dst: cellRef }),
+  })
+  .transform(({ TestLessThan: { lhs, rhs, dst } }) => ({
+    type: HintName.TestLessThan,
+    lhs,
+    rhs,
+    dst,
+  }));
```

Comment:
hints should be in their own file with their own type def, and only imported here